Drafts/1NNN-SB.md

```diff
@@ -0,0 +1,258 @@
+# Resolution of Template Alias Formal Parameters in Template Functions
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            |                                                                 |
+| Author:         | Stefanos Baziotis (sdi1600105@di.uoa.gr)                        |
+| Implementation: | https://github.com/dlang/dmd/pull/9778 (Prototype)              |
+| Status:         |                                                                 |
+
+
+## Contents
+* [Abstract](#abstract)
+* [Description](#description)
+* [Breaking Changes and Deprecations](#breaking-changes-and-deprecations)
+* [Reference](#reference)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Abstract
+
+D has type aliases that can be templates:
+```
+struct TemplateType(T) { }
+alias TemplateAlias(T) = TemplateType!T
+```
+
+Moreover, D has template functions, whose template parameters are inferred / matched when the function is called:
+```
+struct TemplateType(T) { }
+void templateFunction(TemplateType!T arg) { }
+TemplateType!int inst;
+templateFunction(inst); /* template instantiated with T = int */
+```
+
+Combining template aliases and template functions, we would want to be able to have template aliases
+as formal parameter types of template functions:
+```
+struct TemplateType(T) { }
+alias TemplateAlias(T) = TemplateType!T
+void templateFunction(TemplateAlias!T arg) { }
+```
+
+However, the compiler produces an error when such a function has a template alias as a formal parameter.
+This is not a bug, because the desired behaviour is not specified. Thus, although there are Bugzilla reports 
+(see links below), these are tagged as 'Enhancements'.
+
+This DIP proposes a specification of how template alias formal parameters in template functions should be handled.
+
+## Rationale
```

Comment:
Please update the Rationale based on the examples in the latest conversation started at https://github.com/dlang/dmd/pull/9778#issuecomment-496169602. Hope this will help to promote the DIP

Reply:
I'm thinking that based on this conversation, it's good to update why having that feature makes for clearer code than the current alternatives (referenced in the comments). Is this what you thought too?

Reply:
Yes